File: 조성원/실패율.js
const solution = (N, stages) => {
  const fail = Array(N).fill(0);
  const challenge = Array(N).fill(0);

  for (let i = 0; i < N; i++) {
    for (const stage of stages) {
      if (stage >= i + 1) challenge[i]++;
      if (stage === i + 1) fail[i]++;
    }
  }

  const answer = challenge.map((v, i) => ({ stage: i + 1, rate: fail[i] / v }));

  return answer.sort((a, b) => b.rate - a.rate).map((v) => v.stage);
};

File: 김현경/실패율.py
def solution(N, stages):
    fail_rate = []
    user = len(stages)
    for i in range(1, N+1):
        if user != 0:
            fail_rate.append((i, stages.count(i) / user))
            user -= stages.count(i)
        else: # 남는 유저 수가 0인 경우 처리 (ZeroDivisionError 방지)
            fail_rate.append((i, 0))

    fail_rate.sort(key=lambda x: x[1], reverse=True)
    answer = [i[0] for i in fail_rate]
    
    return answer